lib/Discount.php

```diff
@@ -1,23 +1,28 @@
- * @property \Stripe\Coupon $coupon Hash describing the coupon applied to create this discount.
- * @property string|\Stripe\Customer $customer The ID of the customer associated with this discount.
- * @property null|int $end If the coupon has a duration of repeating, the date that this discount will end. If the coupon has a duration of once or forever, this attribute will be null.
- * @property string $id The ID of the discount object.
- * @property null|string $invoice The invoice that the discount’s coupon was applied to, if it was applied directly to a particular invoice.
- * @property null|string $invoice_item The invoice item id (or invoice line item id for invoice line items of type=‘subscription’) that the discount’s coupon was applied to, if it was applied directly to a particular invoice item or invoice line item.
- * @property string $object String representing the object’s type. Objects of the same type share the same value.
- * @property null|string $promotion_code The promotion code applied to create this discount.
+ * @property \Stripe\Coupon $coupon A coupon contains information about a percent-off or amount-off discount you might want to apply to a customer. Coupons may be applied to <a href="https://stripe.com/docs/api#subscriptions">subscriptions</a>, <a href="https://stripe.com/docs/api#invoices">invoices</a>, <a href="https://stripe.com/docs/api/checkout/sessions">checkout sessions</a>, <a href="https://stripe.com/docs/api#quotes">quotes</a>, and more. Coupons do not work with conventional one-off <a href="https://stripe.com/docs/api#create_charge">charges</a> or <a href="https://stripe.com/docs/api/payment_intents">payment intents</a>.
+ * @property null|string|\Stripe\Customer $customer The ID of the customer associated with this discount.
+ * @property null|int $end If the coupon has a duration of <code>repeating</code>, the date that this discount will end. If the coupon has a duration of <code>once</code> or <code>forever</code>, this attribute will be null.
+ * @property null|string $invoice The invoice that the discount's coupon was applied to, if it was applied directly to a particular invoice.
+ * @property null|string $invoice_item The invoice item <code>id</code> (or invoice line item <code>id</code> for invoice line items of type='subscription') that the discount's coupon was applied to, if it was applied directly to a particular invoice item or invoice line item.
+ * @property null|string|\Stripe\PromotionCode $promotion_code The promotion code applied to create this discount.
  * @property int $start Date that the coupon was applied.
  * @property null|string $subscription The subscription that this coupon is applied to, if it is applied to a particular subscription.
  */
-class Discount extends StripeObject
+class Discount extends ApiResource
```

Comment:
Is this correct/intentional?

This causes `Discount` to have e.g. a `.refresh()` method and an `.instance_url` method and I'm not sure that's intended.

Reply:
Yes, all resources have ApiResource base class in PHP. See [UsageRecordSummary](https://github.com/stripe/stripe-php/blob/7d0a90772fc1c179e370971264318208533324b9/lib/UsageRecordSummary.php#L16) for example.